### ISSUE TITLE
broken bar

### ISSUE THREAD

**Issue by Dialgatrainer02** [end-4/dots-hyprland#47]
running eww open bar results in the bar showing then disapearing 
### eww logs
━━━━━━━━━━━━━━━━━━━━━━━┓
┃Initializing eww daemon┃
┗━━━━━━━━━━━━━━━━━━━━━━━┛
    
 2023-08-13T12:05:04.684Z INFO  eww::app    > Opening window bar
 2023-08-13T12:05:04.686Z INFO  eww::ipc_server > IPC server initialized
 2023-08-13T12:05:04.915Z WARN  eww::script_var_handler > stderr of `notifications`: Error: name already taken on the bus
 2023-08-13T12:05:04.927Z WARN  eww::script_var_handler > stderr of `lang_ibus`: sh: line 1: ibus: command not found
 2023-08-13T12:05:04.930Z WARN  eww::script_var_handler > stderr of `lang_ibus`: /usr/src/debug/gcc/gcc-build/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.h:2119: void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::pop_back() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]: Assertion '!empty()' failed.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: BoolError { message: "property 'value' of type 'CircProg' can't be set from given value, it is invalid or out of range", filename: "/home/olivia/eww/XDG_DATA_HOME/cargo/registry/src/github.com-1ecc6299db9ec823/glib-0.15.12/src/object.rs", function: "glib::object", line: 3393 }', /home/olivia/eww/XDG_DATA_HOME/cargo/registry/src/github.com-1ecc6299db9ec823/glib-0.15.12/src/object.rs:2238:53
stack backtrace:
   0:     0x55717123d8d0 - std::backtrace_rs::backtrace::libunwind::trace::ha93b1d4802475003
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x55717123d8d0 - std::backtrace_rs::backtrace::trace_unsynchronized::h40d7f2da011bd85b
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55717123d8d0 - std::sys_common::backtrace::_print_fmt::hd4c318607ed6977d
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x55717123d8d0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3d1dee5bf49c390e
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x55717126496e - core::fmt::write::h5629927f37465513
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/core/src/fmt/mod.rs:1202:17
   5:     0x5571712390c5 - std::io::Write::write_fmt::h41e38f576096734b
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/io/mod.rs:1672:15
   6:     0x55717123f143 - std::sys_common::backtrace::_print::hb4df1a69959f2fbc
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x55717123f143 - std::sys_common::backtrace::print::h7093396eb4e737fc
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x55717123f143 - std::panicking::default_hook::{{closure}}::h044155a56acd855d
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:295:22
   9:     0x55717123ee2f - std::panicking::default_hook::hf2b314c3c2ce4158
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:314:9
  10:     0x55717123f7ea - std::panicking::rust_panic_with_hook::he1240b7f7e10f2ad
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:698:17
  11:     0x55717123f6e7 - std::panicking::begin_panic_handler::{{closure}}::h60ca41cdf195e052
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:588:13
  12:     0x55717123ddec - std::sys_common::backtrace::__rust_end_short_backtrace::h0260dedd111c7b11
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x55717123f402 - rust_begin_unwind
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:584:5
  14:     0x557170b10bc3 - core::panicking::panic_fmt::h44c2814e5bc1b46f
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/core/src/panicking.rs:142:14
  15:     0x557170b10d13 - core::result::unwrap_failed::h09f85debf7d2260a
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/core/src/result.rs:1785:5
  16:     0x557170b82cb0 - <T as glib::object::ObjectExt>::set_property::haa6ac52437a15477
  17:     0x557170c7e425 - eww::widgets::widget_definitions::build_circular_progress_bar::{{closure}}::h0af032204ce40596
  18:     0x557170b48e18 - eww::state::scope_graph::ScopeGraph::notify_value_changed::hcb37e32a60790b06
  19:     0x557170b48f58 - eww::state::scope_graph::ScopeGraph::notify_value_changed::hcb37e32a60790b06
  20:     0x557170b4876b - eww::state::scope_graph::ScopeGraph::update_value::h4679a8b8128c96e8
  21:     0x557170bbb8d7 - eww::app::App<B>::handle_command::h1efaec93c8e027d2
  22:     0x557170be314e - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::ha4606adc63ec8bd1
  23:     0x5571712190bc - glib::main_context::<impl glib::auto::main_context::MainContext>::with_thread_default::hf524b2fdada48284
  24:     0x55717121c56a - glib::main_context_futures::TaskSource::dispatch::h114fe23fc94442ac
  25:     0x7f85a385fa31 - g_main_context_dispatch
  26:     0x7f85a38bccc9 - <unknown>
  27:     0x7f85a385efef - g_main_loop_run
  28:     0x7f85a3fecf6f - gtk_main
  29:     0x557170b45bca - eww::server::initialize_server::hff660dd79df03d40
  30:     0x557170ce81ae - eww::main::h9eaad2179cee480b
  31:     0x557170c84dd3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h279b4285ba55781b
  32:     0x557170c85b09 - std::rt::lang_start::{{closure}}::hd8f10a894ed9311e
  33:     0x5571712326cf - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h1f73150d454b41b9
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/core/src/ops/function.rs:280:13
  34:     0x5571712326cf - std::panicking::try::do_call::hc2a0f6a54bbfbc83
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:492:40
  35:     0x5571712326cf - std::panicking::try::h22dd2c39c33f5d79
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:456:19
  36:     0x5571712326cf - std::panic::catch_unwind::h304ac8613501ec27
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panic.rs:137:14
  37:     0x5571712326cf - std::rt::lang_start_internal::{{closure}}::h908b7e245713b406
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/rt.rs:128:48
  38:     0x5571712326cf - std::panicking::try::do_call::h1343d879ac83b3c6
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:492:40
  39:     0x5571712326cf - std::panicking::try::he03f984691a7ab4b
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panicking.rs:456:19
  40:     0x5571712326cf - std::panic::catch_unwind::h01722b96585f9bcd
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/panic.rs:137:14
  41:     0x5571712326cf - std::rt::lang_start_internal::h25ef46d553ec36b4
                               at /rustc/c07a8b4e09f356c7468b69c50cac7fc5b5000b8a/library/std/src/rt.rs:128:20
  42:     0x557170ce92b2 - main
  43:     0x7f85a3427cd0 - <unknown>
  44:     0x7f85a3427d8a - __libc_start_main
  45:     0x557170b10f15 - _start
  46:                0x0 - <unknown>
 this is from running eww open bar with backtrace to full

**Comment by end-4** [comment]
- which branch is this?
- from what i can see from this messy log:
  - try to kill any other running notification daemon (dunst/mako/etc) before launching
  - install ibus
  - crazy rust errors that i'm seriously unable to solve

**Comment by Dialgatrainer02** [comment]
with ibus installed and starting it has remove the ibus errors but i still get this warning stderr of `notifications`: Error: name already taken on the bus
and a bunch of other rust errors and it continuing to crash

**Comment by end-4** [comment]
> try to kill any other running notification daemon (dunst/mako/etc) before launching

ahem

**Comment by Dialgatrainer02** [comment]
i forgot to mention i also did that

**Comment by end-4** [comment]
- idk what's happening with that long ibus gcc error.... are you on ARM or anything?
- about the rust stacktrace, i'm sorry but i can't do anything about that

Maybe try to compile eww and the ibus script (`~/.config/eww/scripts/language.cpp`) again?

**Comment by end-4** [comment]
also which branch

**Comment by Dialgatrainer02** [comment]
novelknock using trig pr eww (i updated that before i made an issue here) im using an x86 laptop and it broke after an update i think

**Comment by Dialgatrainer02** [comment]
for the backtrace should i open an issue with eww directly?

**Comment by end-4** [comment]
does it throw errors when you use normal eww?

**Comment by Dialgatrainer02** [comment]
same error on latest eww from the github

**Comment by Dialgatrainer02** [comment]
ound the rust backtrace issue here https://github.com/elkowar/eww/issues/854 (im not sure how to mention issues outside of the current repo so here is  a link)

**Comment by end-4** [comment]
that should be fixed really easy
just write something like `${myvar > 100 ? 100 : myvar}`

**Comment by Dialgatrainer02** [comment]
this hasn't fixed it and i am getting the exact same error

**Comment by Dialgatrainer02** [comment]
i applied this to swap cpu memory and battery from the commit and it's still failing with that error

**Comment by end-4** [comment]
- they aren't sourced from the same place
- the sideleft widget is just in `~/.config/eww/windows/sideleft.yuck`
- the bar
  - window declaration is in `~/.config/eww/eww.yuck`
  - it uses modules in `~/.config/eww/modules/` folder

**Comment by Dialgatrainer02** [comment]
When I get access to my laptop I'll try and set the max for the bar values

**Comment by MyLiege23** [comment]
Hi y'all. I'm having the same symptoms. I have a different setup, I compiled with eww-wayland via the AUR. Just a few hours ago, everything was working. I've also compiled eww from the trig branch (looks like it merged to main today - correct me if I'm wrong) compiled and run with ./eww open bar etc. I used rustup nightly and installed everything today, so everything should be up to date. I'm also using the novelknock branch.

```
┏━━━━━━━━━━━━━━━━━━━━━━━┓
┃Initializing eww daemon┃
┗━━━━━━━━━━━━━━━━━━━━━━━┛

 2023-08-16T22:40:45.421Z WARN  eww::config::system_stats > Failed to get/calculate uWh: the total_avg value of the battery magic var will probably be a garbage value that can not be trusted.
 2023-08-16T22:40:45.429Z INFO  eww::app                  > Opening window bar
 2023-08-16T22:40:45.429Z INFO  eww::ipc_server           > IPC server initialized
 2023-08-16T22:40:45.500Z WARN  eww::script_var_handler   > stderr of `brightness`: scripts/brightness: line 50: light: command not found
 2023-08-16T22:40:45.500Z WARN  eww::script_var_handler   > stderr of `brightness`: scripts/brightness: line 51: light: command not found
 2023-08-16T22:40:45.509Z WARN  eww::script_var_handler   > stderr of `notifications`: Error: name already taken on the bus
 2023-08-16T22:40:45.518Z WARN  eww::script_var_handler   > stderr of `battery`: cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
 2023-08-16T22:40:45.523Z WARN  eww::script_var_handler   > stderr of `battery`: (standard_in) 1: syntax error
 2023-08-16T22:40:45.524Z WARN  eww::script_var_handler   > stderr of `battery`: cat: '/sys/class/power_supply/*/charge_full': No such file or directory
 2023-08-16T22:40:45.525Z WARN  eww::script_var_handler   > stderr of `battery`: cat: '/sys/class/power_supply/*/charge_now': No such file or directory
 2023-08-16T22:40:45.526Z WARN  eww::script_var_handler   > stderr of `battery`: scripts/battery: line 9: [: : integer expression expected
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: BoolError { message: "property 'value' of type 'CircProg' can't be set from given value, it is invalid or out of range", filename: "/home/myliege/.cargo/registry/src/github.com-1ecc6299db9ec823/glib-0.15.12/src/object.rs", function: "glib::object", line: 3393 }', /home/myliege/.cargo/registry/src/github.com-1ecc6299db9ec823/glib-0.15.12/src/object.rs:2238:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I'm on a desktop with no battery, hence the battery warnings.  I also have errors with the brightness control because light doesn't work with ddc control on external monitors (to my knowledge), but I commented out the lines in the osd_bright.yuck config for now that give those errors. Hope this helps troubleshoot! I'm wondering if there's a bug that toggles the visibility of the bar, because when the bar shows up, it fades away rather than just disappearing, and the shortcuts for the side bars/powerview/music dock all work as expected.

**Comment by end-4** [comment]
@MyLiege23 
- did you go through the dependencies list? `light` command is missing
- kill any other notification before starting eww so that notification works

**Comment by end-4** [comment]
can you guys run the battery script from the commandline
`cd ~/.config/eww && scripts/battery`

**Comment by MyLiege23** [comment]
Absolutely :)

```
cd ~/.config/eww && scripts/battery                                                                                                                                                                      ─╯
css/  eww_covers/  images/  json/  modules/  scripts/  windows/  eww.scss*  eww.yuck*  variables.yuck*
cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
(standard_in) 1: syntax error
cat: '/sys/class/power_supply/*/charge_full': No such file or directory
cat: '/sys/class/power_supply/*/charge_now': No such file or directory
scripts/battery: line 9: [: : integer expression expected
{"quickicon": "", "icon": "", "percentage": 33, "wattage": " W", "status": "h left", "circolor": "#ffdad6", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
(standard_in) 1: syntax error
cat: '/sys/class/power_supply/*/charge_full': No such file or directory
cat: '/sys/class/power_supply/*/charge_now': No such file or directory
scripts/battery: line 9: [: : integer expression expected
{"quickicon": "", "icon": "", "percentage": 33, "wattage": " W", "status": "h left", "circolor": "#ffdad6", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
(standard_in) 1: syntax error
cat: '/sys/class/power_supply/*/charge_full': No such file or directory
cat: '/sys/class/power_supply/*/charge_now': No such file or directory
scripts/battery: line 9: [: : integer expression expected
{"quickicon": "", "icon": "", "percentage": 33, "wattage": " W", "status": "h left", "circolor": "#ffdad6", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
(standard_in) 1: syntax error
cat: '/sys/class/power_supply/*/charge_full': No such file or directory
cat: '/sys/class/power_supply/*/charge_now': No such file or directory
scripts/battery: line 9: [: : integer expression expected
{"quickicon": "", "icon": "", "percentage": 33, "wattage": " W", "status": "h left", "circolor": "#ffdad6", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
cat: '/sys/class/power_supply/*/voltage_now': No such file or directory
^C
```

my script loops with a syntax error because I have no battery. There are a few other bugs as well. Light doesn't work for changing my monitor brightness - I am trying to implement ddcci-driver-linux-dkms to solve that problem, but I haven't got it working quite yet.

**Comment by end-4** [comment]
okay. @MyLiege23 have you tried running `usermod -aG video <USERNAME>` ? `light` should work after that. if you have any other questions please open a new issue

waiting for @Dialgatrainer02 's battery script run result

**Comment by Dialgatrainer02** [comment]
{"quickicon": "", "icon": "", "percentage": 81, "wattage": "11.9 W", "status": "00:14h left", "circolor": "#ffdcbf", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
{"quickicon": "", "icon": "", "percentage": 81, "wattage": "11.8 W", "status": "00:14h left", "circolor": "#ffdcbf", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
{"quickicon": "", "icon": "", "percentage": 81, "wattage": "11.9 W", "status": "00:14h left", "circolor": "#ffdcbf", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
{"quickicon": "", "icon": "", "percentage": 81, "wattage": "11.9 W", "status": "00:14h left", "circolor": "#ffdcbf", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
{"quickicon": "", "icon": "", "percentage": 81, "wattage": "11.9 W", "status": "00:14h left", "circolor": "#ffdcbf", "cirbgcolor": "transparent", "color": "white", "bgcolor": "transparent" }
this is the output of that

**Comment by end-4** [comment]
percentage 81 is fine
i checked for any other circular progress and made sure they don't go over 100

**Comment by MyLiege23** [comment]
@end-4 thanks I'm not sure whether that fixed the brightness issue, but that's not in the scope of this issue thread. I'll open another issue if I'm sure that it's related to this git repo. And I'm not getting an error from eww about bright, so I'll be adding your suggestion to my own dots (which will be an extension of yours with my own programs like steam, etc). 

I'm going to do a fresh install of endeavourOS gnome and retry your dots to see if the eww bar disappearing persists. It might be an issue from my setup. I'll report here if the problem happens again.

**Comment by Dialgatrainer02** [comment]
im updating to the latest eww git now the trig pr has been merged and ill see if that fixes it

**Comment by Dialgatrainer02** [comment]
the bar works now i think the trig version was out of date but radial menu still isnt working regardless of trig or not

**Comment by end-4** [comment]
cool
the feature got merged btw, just compile eww from upstream

**Comment by end-4** [comment]
with the new eww my setup works normally
